src/pages/home/report/ContextMenu/BaseReportActionContextMenu.js

```diff
@@ -41,94 +41,85 @@ const defaultProps = {
     ...GenericReportActionContextMenuDefaultProps,
 };
-class BaseReportActionContextMenu extends React.Component {
-    constructor(props) {
-        super(props);
-        this.wrapperStyle = getReportActionContextMenuStyles(this.props.isMini, this.props.isSmallScreenWidth);
-
-        this.state = {
-            shouldKeepOpen: false,
-        };
-    }
-
-    render() {
-        const shouldShowFilter = (contextAction) =>
-            contextAction.shouldShow(
-                this.props.type,
-                this.props.reportAction,
-                this.props.isArchivedRoom,
-                this.props.betas,
-                this.props.anchor,
-                this.props.isChronosReport,
-                this.props.reportID,
-                this.props.isPinnedChat,
-                this.props.isUnreadChat,
-            );
-
-        /**
-         * Checks if user is anonymous. If true and the action doesn't accept for anonymous user, hides the context menu and
-         * shows the sign in modal. Else, executes the callback.
-         *
-         * @param {Function} callback
-         * @param {Boolean} isAnonymousAction
-         */
-        const interceptAnonymousUser = (callback, isAnonymousAction = false) => {
-            if (Session.isAnonymousUser() && !isAnonymousAction) {
-                hideContextMenu(false);
-
-                InteractionManager.runAfterInteractions(() => {
-                    Session.signOutAndRedirectToSignIn();
-                });
-            } else {
-                callback();
-            }
-        };
-
-        return (
-            (this.props.isVisible || this.state.shouldKeepOpen) && (
-                <View
-                    ref={this.props.contentRef}
-                    style={this.wrapperStyle}
-                >
-                    {_.map(_.filter(ContextMenuActions, shouldShowFilter), (contextAction) => {
-                        const closePopup = !this.props.isMini;
-                        const payload = {
-                            reportAction: this.props.reportAction,
-                            reportID: this.props.reportID,
-                            draftMessage: this.props.draftMessage,
-                            selection: this.props.selection,
-                            close: () => this.setState({shouldKeepOpen: false}),
-                            openContextMenu: () => this.setState({shouldKeepOpen: true}),
-                            interceptAnonymousUser,
-                        };
-
-                        if (contextAction.renderContent) {
-                            // make sure that renderContent isn't mixed with unsupported props
-                            if (__DEV__ && (contextAction.text != null || contextAction.icon != null)) {
-                                throw new Error('Dev error: renderContent() and text/icon cannot be used together.');
-                            }
+function BaseReportActionContextMenu(props) {
```

Comment:
Can we destructure the props here?
As per this line of `STYLE.md`:
[Always use destructuring to get prop values. Destructuring is necessary to assign default values to props. ](https://github.com/robertKozik/Expensify_app_fork/blob/c24579548097a7cb0dfc2093052086f4c01251ed/contributingGuides/STYLE.md?plain=1#L213)

Reply:
@robertKozik I tried to use destructuring to get prop values but I am getting this lint errors. It requires to use default props.
 
![Screenshot 2023-08-01 at 21 58 21](https://github.com/Expensify/App/assets/138146362/d80e3978-8e3c-4c32-b38a-678c216a578d)

Reply:
Can't we assign default values to make lint happy?

Reply:
I checked other fc components but they don't use destructuring as well.

Reply:
Okay, got it, thank you